File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = ['' for _ in range(n)]
    
    # 1. binary
    for i in range(n):    
        arr1[i] = format(arr1[i], 'b').zfill(n)
        arr2[i] = format(arr2[i], 'b').zfill(n)
    
    
    # 2. answer
    for i in range(n):
        for j in range(n):
            if arr1[i][j] == '1' and arr2[i][j] == '1':
                answer[i] += '#'
            
            elif arr1[i][j] == '0' and arr2[i][j] == '0':
                answer[i] += ' '
            
            else:
                answer[i] += '#'
   

    return answer

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        String makeCode;
        
        for (int i = 0; i < n; i++) {
            // 하나라도 1이면 1
            // str.length()가 N보다 작을 경우 공백 추가
            makeCode = String.format("%16s", Integer.toBinaryString(arr1[i] | arr2[i]));
            makeCode = makeCode.substring(makeCode.length() - n);
            makeCode = makeCode.replaceAll("1", "#");
            makeCode = makeCode.replaceAll("0", " ");            
            answer[i] = makeCode;
        }
        
        return answer;
    }
}